content/post/2021-06-28-draw-me-a-project/index.md

```diff
@@ -0,0 +1,267 @@
+
+{{< tweet 1403009688392781824 >}}
+
+As the first part of my talk really is a collection of awesome content by Jenny Bryan, I will also show her rules for [naming files](http://www2.stat.duke.edu/~rcs46/lectures_2015/01-markdown-git/slides/naming-slides/naming-slides.pdf). What Shakespeare said about [roses](https://en.wikipedia.org/wiki/A_rose_by_any_other_name_would_smell_as_sweet) isn't true when programming. 😅
+
+Then I'll "introduce" version control. I'll briefly mention some basic git commands, and most importantly, how to run them! Here are my preferences:
+
+-   Using usethis ([`usethis::use_git()`](https://usethis.r-lib.org/reference/use_git.html), [`usethis::use_github()`](https://usethis.r-lib.org/reference/use_github.html), etc.) and gert ([`gert::git_push()`](https://docs.ropensci.org/gert/reference/git_fetch.html)) so that I don't need to leave R.
+
+-   Using RStudio git pane.
+
+-   Using the terminal for commands copy-pasted from somewhere on the web (and for the ones I now know by heart!).
+
+-   Using a git interface like GitKraken for more complicated stuff... which I haven't really ever fully explored, actually.
+
+I'll share links to my favorite git resources that I send to anyone who asks (or doesn't ask):
+
+-   [Excuse Me, Do You Have a Moment to Talk About Version Control?](https://www.tandfonline.com/doi/full/10.1080/00031305.2017.1399928), Jenny Bryan.
+
+-   [Happy Git and GitHub for the useR](https://happygitwithr.com/), Jenny Bryan, the STAT 545 TAs, Jim Hester.
+
+{{< tweet 970383613119184896 >}}
+
+-   [Reflections on 4 months of GitHub: my advice to beginners](https://suzan.rbind.io/2018/03/reflections-4-months-of-github/), Suzan Baert.
+
+## How to protect your project from external changes
+
+Here's a [maximum credible accident](https://en.wikipedia.org/wiki/Design-basis_event): you write some pretty and handy code munging your data using `package::my_favorite_function()`. Now you go and update that package and realize `my_favorite_function` is gone! It was apparently removed for good reasons but now your script is broken!
+
+To prevent that, you need to encapsulate your project. You can track and restore package dependencies of your package by using the [renv package](https://rstudio.github.io/renv/) by Kevin Ushey. The renv package is the successor of the packrat package by the same author.
+
+Using renv is actually quite easy:
+
+-   In a new project you run [`renv::init()`](https://rstudio.github.io/renv//reference/init.html);
+-   After that you install and remove packages as you normally would (renv is smart and will copy files from your local not-project library to be faster). Metadata about packages (where do they come from) are stored in the `renv.lock` file that you'd put under version control;
```

Comment:
This require a manual `renv::snapshot()` but I think you know. 

Unless auto snapshot is configured but that can be quite harsh.

Reply:
good catch :grimacing: